src/apyfixed.cc

```diff
@@ -973,6 +982,49 @@ APyFixed APyFixed::from_double(
     return result;
 }
 
+APyFixed APyFixed::from_integer(
+    const nb::int_& value,
+    std::optional<int> int_bits,
+    std::optional<int> frac_bits,
+    std::optional<int> bits
+)
+{
+    // Extract things bit widths
+    const int res_bits = bits_from_optional(bits, int_bits, frac_bits);
+    const int res_int_bits = int_bits.has_value() ? *int_bits : *bits - *frac_bits;
+
+    APyFixed result(res_bits, res_int_bits);
+    result._data = python_long_to_limb_vec(value, result._data.size());
+
+    // Adjust the number
+    if (result.frac_bits() > 0) {
+        limb_vector_lsl(result._data.begin(), result._data.end(), result.frac_bits());
+    } else { /* result.frac_bits() <= 0 */
+        limb_vector_asr(result._data.begin(), result._data.end(), -result.frac_bits());
+    }
+
+    // Two's-complements overflow bits outside of the range
+    result._overflow_twos_complement(
+        result._data.begin(), result._data.end(), res_bits, res_int_bits
+    );
+
+    return result;
+}
+
+APyFixed APyFixed::from_unspecified_integer(const nb::int_& value)
+{
+    std::vector<mp_limb_t> limbs = python_long_to_limb_vec(value);
+
+    // If the value is positive but the MSB is set, adding a zero limb
+    // will prevent it from representing a negative value.
+    if (!PyLong_IsNegative((const PyLongObject*)value.ptr())
```

Comment:
The build is failing for Python 3.12 due to this line. The error message is:
 `../src/python_util.h:55:39: error: ‘PyObject’ {aka ‘struct _object’} has no member named ‘long_value’`
 Do you know how to fix this @miklhh? It seems like it's not finding the correct version of `PyLong_IsNegative`, but the macro is also used in `python_long_to_limb_vec` and it has not complained before.

Reply:
> ## Order of the conversions
> 
> Implicit conversion sequence consists of the following, in this order:
> 1) zero or one standard conversion sequence;
> 2) zero or one user-defined conversion;
> 3) zero or one standard conversion sequence (only if a user-defined conversion is used).

\* https://en.cppreference.com/w/cpp/language/implicit_conversion